Modules/Core/Common/src/itkObject.cxx

```diff
@@ -112,42 +113,35 @@ class ITKCommon_HIDDEN SubjectImplementation
     bool                    m_Save;
   };
 
+  using ObserverListType = std::list<std::unique_ptr<const Observer>>;
+
   template <typename TObject>
   void
-  InvokeEventRecursion(const EventObject & event, TObject * self, std::list<Observer *>::reverse_iterator & i);
+  InvokeEventRecursion(const EventObject & event, TObject * self, ObserverListType::reverse_iterator & i);
```

Comment:
Will this change in the public interface cause backwards compatibility issues?

Reply:
Good question, thanks @dzenanz  but no, I don't see any backwards compatibility issue here. `InvokeEventRecursion` is a protected member function of a "hidden" class (`ITKCommon_HIDDEN`) named "SubjectImplementation", which is defined inside itkObject.cxx. So `InvokeEventRecursion` is also declared and defined in itkObject.cxx, and it isn't being called outside itkObject.cxx. So it's really just an implementation detail. As far as I can see.